Demo app ICSharpCode.AvalonEdit error

building the source code then using the demo app by going to certain controls will cause exception
 
`System.Windows.Markup.XamlParseException: 'Could not load file or assembly 'ICSharpCode.AvalonEdit, Version=6.0.178.0, Culture=neutral, PublicKeyToken=9cc39be672370310' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)'
`
`FileLoadException: Could not load file or assembly 'ICSharpCode.AvalonEdit, Version=6.0.178.0, Culture=neutral, PublicKeyToken=9cc39be672370310' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)` 

the file is there and that error for the dot net core and framework versions

~~~
C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit> dotnet build MaterialDesignToolkit.Wpf.sln
Microsoft (R) Build Engine version 16.3.0+0f4c62fea for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 296.04 ms for C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MahMaterialDragablzMashUp\MahAppsDragablzDemo.csproj.
  Restore completed in 295.75 ms for C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignColors.Wpf\MaterialDesignColors.Wpf.csproj.
  Restore completed in 295.88 ms for C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MainDemo.Wpf\MaterialDesignDemo.csproj.
  Restore completed in 23.48 ms for C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.Wpf\MaterialDesignThemes.Wpf.csproj.
C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.MahApps\MaterialDesignThemes.MahApps.csproj : warning NU1701: Package 'ControlzEx 3.0.2.4' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.0'. This package may not be fully compatible with your project. [C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignToolkit.Wpf.sln]
C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.MahApps\MaterialDesignThemes.MahApps.csproj : warning NU1701: Package 'MahApps.Metro 1.6.5' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.0'. This package may not be fully compatible with your project. [C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignToolkit.Wpf.sln]
  Restore completed in 761.93 ms for C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.MahApps\MaterialDesignThemes.MahApps.csproj.
  Restore completed in 907.81 ms for C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.Wpf.Tests\MaterialDesignThemes.Wpf.Tests.csproj.
  Restore completed in 1.27 sec for C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignColors.Wpf.Tests\MaterialDesignColors.Wpf.Tests.csproj.
C:\Program Files\dotnet\sdk\3.0.100\Sdks\Microsoft.NET.Sdk.WindowsDesktop\targets\Microsoft.NET.Sdk.WindowsDesktop.targets(96,5): warning NETSDK1106: Microsoft.NET.Sdk.WindowsDesktop requires 'UseWpf' or 'UseWindowsForms' to be set to 'true' [C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MainDemo.Wpf\MaterialDesignDemo.csproj]
C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.MahApps\MaterialDesignThemes.MahApps.csproj : warning NU1701: Package 'ControlzEx 3.0.2.4' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.0'. This package may not be fully compatible with your project.
C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.MahApps\MaterialDesignThemes.MahApps.csproj : warning NU1701: Package 'MahApps.Metro 1.6.5' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.0'. This package may not be fully compatible with your project.
C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.MahApps\MaterialDesignThemes.MahApps.csproj : warning NU1701: Package 'ControlzEx 3.0.2.4' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.0'. This package may not be fully compatible with your project.
C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.MahApps\MaterialDesignThemes.MahApps.csproj : warning NU1701: Package 'MahApps.Metro 1.6.5' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.0'. This package may not be fully compatible with your project.
C:\Program Files\dotnet\sdk\3.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(347,5): warning NETSDK1107: Microsoft.NET.Sdk.WindowsDesktop is required to build Windows desktop applications. 'UseWpf' and 'UseWindowsForms' are not supported by the current SDK. [C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.Wpf.Tests\MaterialDesignThemes.Wpf.Tests.csproj]
C:\Program Files\dotnet\sdk\3.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(347,5): warning NETSDK1107: Microsoft.NET.Sdk.WindowsDesktop is required to build Windows desktop applications. 'UseWpf' and 'UseWindowsForms' are not supported by the current SDK. [C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.Wpf.Tests\MaterialDesignThemes.Wpf.Tests.csproj]
  MaterialDesignColors.Wpf -> C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignColors.Wpf\bin\Debug\net45\MaterialDesignColors.dll
  MaterialDesignColors.Wpf -> C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignColors.Wpf\bin\Debug\netcoreapp3.0\MaterialDesignColors.dll
  MaterialDesignColors.Wpf.Tests -> C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignColors.Wpf.Tests\bin\Debug\net45\MaterialDesignColors.Wpf.Fixture.dll
  MaterialDesignThemes.Wpf -> C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.Wpf\bin\Debug\net45\MaterialDesignThemes.Wpf.dll
  MaterialDesignThemes.Wpf.Tests -> C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.Wpf.Tests\bin\Debug\net472\MaterialDesignThemes.Wpf.Tests.dll
  MaterialDesignThemes.MahApps -> C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.MahApps\bin\Debug\net45\MaterialDesignThemes.MahApps.dll
  MaterialDesignThemes.Wpf -> C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.Wpf\bin\Debug\netcoreapp3.0\MaterialDesignThemes.Wpf.dll
C:\Program Files\dotnet\sdk\3.0.100\Microsoft.Common.CurrentVersion.targets(2106,5): warning MSB3277: Found conflicts between different versions of "ICSharpCode.AvalonEdit" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MainDemo.Wpf\MaterialDesignDemo.csproj]
  MahAppsDragablzDemo -> C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MahMaterialDragablzMashUp\bin\Debug\net45\MahAppsDragablzDemo.exe
  MaterialDesignColors.Wpf.Tests -> C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignColors.Wpf.Tests\bin\Debug\netcoreapp3.0\MaterialDesignColors.Wpf.Fixture.dll
C:\Program Files\dotnet\sdk\3.0.100\Microsoft.Common.CurrentVersion.targets(2106,5): warning MSB3277: Found conflicts between different versions of "ICSharpCode.AvalonEdit" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MainDemo.Wpf\MaterialDesignDemo.csproj]
C:\Program Files\dotnet\sdk\3.0.100\Microsoft.Common.CurrentVersion.targets(2106,5): warning MSB3277: Found conflicts between different versions of "System.IO.Pipes.AccessControl" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MainDemo.Wpf\MaterialDesignDemo.csproj]
  MaterialDesignDemo -> C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MainDemo.Wpf\bin\Debug\net472\MaterialDesignDemo.exe
  MaterialDesignThemes.MahApps -> C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.MahApps\bin\Debug\netcoreapp3.0\MaterialDesignThemes.MahApps.dll
  MaterialDesignThemes.Wpf.Tests -> C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.Wpf.Tests\bin\Debug\netcoreapp3.0\MaterialDesignThemes.Wpf.Tests.dll
  MaterialDesignDemo -> C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MainDemo.Wpf\bin\Debug\netcoreapp3.0\MaterialDesignDemo.dll

Build succeeded.

C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.MahApps\MaterialDesignThemes.MahApps.csproj : warning NU1701: Package 'ControlzEx 3.0.2.4' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.0'. This package may not be fully compatible with your project. [C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignToolkit.Wpf.sln]
C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.MahApps\MaterialDesignThemes.MahApps.csproj : warning NU1701: Package 'MahApps.Metro 1.6.5' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.0'. This package may not be fully compatible with your project. [C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignToolkit.Wpf.sln]
C:\Program Files\dotnet\sdk\3.0.100\Sdks\Microsoft.NET.Sdk.WindowsDesktop\targets\Microsoft.NET.Sdk.WindowsDesktop.targets(96,5): warning NETSDK1106: Microsoft.NET.Sdk.WindowsDesktop requires 'UseWpf' or 'UseWindowsForms' to be set to 'true' [C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MainDemo.Wpf\MaterialDesignDemo.csproj]
C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.MahApps\MaterialDesignThemes.MahApps.csproj : warning NU1701: Package 'ControlzEx 3.0.2.4' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.0'. This package may not be fully compatible with your project.
C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.MahApps\MaterialDesignThemes.MahApps.csproj : warning NU1701: Package 'MahApps.Metro 1.6.5' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.0'. This package may not be fully compatible with your project.
C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.MahApps\MaterialDesignThemes.MahApps.csproj : warning NU1701: Package 'ControlzEx 3.0.2.4' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.0'. This package may not be fully compatible with your project.
C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.MahApps\MaterialDesignThemes.MahApps.csproj : warning NU1701: Package 'MahApps.Metro 1.6.5' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v3.0'. This package may not be fully compatible with your project.
C:\Program Files\dotnet\sdk\3.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(347,5): warning NETSDK1107: Microsoft.NET.Sdk.WindowsDesktop is required to build Windows desktop applications. 'UseWpf' and 'UseWindowsForms' are not supported by the current SDK. [C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.Wpf.Tests\MaterialDesignThemes.Wpf.Tests.csproj]
C:\Program Files\dotnet\sdk\3.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.targets(347,5): warning NETSDK1107: Microsoft.NET.Sdk.WindowsDesktop is required to build Windows desktop applications. 'UseWpf' and 'UseWindowsForms' are not supported by the current SDK. [C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MaterialDesignThemes.Wpf.Tests\MaterialDesignThemes.Wpf.Tests.csproj]
C:\Program Files\dotnet\sdk\3.0.100\Microsoft.Common.CurrentVersion.targets(2106,5): warning MSB3277: Found conflicts between different versions of "ICSharpCode.AvalonEdit" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MainDemo.Wpf\MaterialDesignDemo.csproj]
C:\Program Files\dotnet\sdk\3.0.100\Microsoft.Common.CurrentVersion.targets(2106,5): warning MSB3277: Found conflicts between different versions of "ICSharpCode.AvalonEdit" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MainDemo.Wpf\MaterialDesignDemo.csproj]
C:\Program Files\dotnet\sdk\3.0.100\Microsoft.Common.CurrentVersion.targets(2106,5): warning MSB3277: Found conflicts between different versions of "System.IO.Pipes.AccessControl" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit\MainDemo.Wpf\MaterialDesignDemo.csproj]
    12 Warning(s)
    0 Error(s)

Time Elapsed 00:00:32.05
PS C:\Users\ahmed\source\repos\MaterialDesignInXamlToolkit>
~~~

--- a/MainDemo.Wpf/App.config
+++ b/MainDemo.Wpf/App.config
@@ -6,4 +6,13 @@
   <appSettings>
     <add key="GitHub" value="https://github.com/ButchersBoy/MaterialDesignInXamlToolkit"/>
   </appSettings>
+  <!-- Can be removed once ShowMeTheXaml.AvalonEdit is updated -->
+  <runtime>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="ICSharpCode.AvalonEdit" publicKeyToken="9cc39be672370310" culture="neutral"/>
+        <bindingRedirect oldVersion="6.0.178.0" newVersion="6.0.0.0"/>
+      </dependentAssembly>
+    </assemblyBinding>
+  </runtime>
 </configuration>

